<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.0.38">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Edward Lau">
<meta name="dcterms.date" content="2025-05-01">
<meta name="keywords" content="Protein turnover, Proteomics, Mass spectrometry">
<meta name="description" content="RIANA enables protein turnover data analysis from mass spectrometry-based proteomics experiments. Supports multiple labeling strategies, including labeled amino acid and heavy water.">

<title>RIANA v0.8.1 - Frequently Asked Questions</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>

  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<link rel="stylesheet" href="styles.css">
<meta property="og:title" content="RIANA v0.8.1 - Frequently Asked Questions">
<meta property="og:description" content="RIANA enables protein turnover data analysis from mass spectrometry-based proteomics experiments. Supports multiple labeling strategies, including labeled amino acid and heavy water.">
<meta property="og:image" content="https://github.com/ed-lau/jcast/images/riana_logo.png">
<meta property="og:site-name" content="RIANA v0.8.1">
<meta property="og:image:height" content="1034">
<meta property="og:image:width" content="2550">
<meta property="og:image:alt" content="RIANA - Relative Isotope Abundance Analyzer">
<meta name="twitter:title" content="RIANA v0.8.1 - Frequently Asked Questions">
<meta name="twitter:description" content="RIANA enables protein turnover data analysis from mass spectrometry-based proteomics experiments. Supports multiple labeling strategies, including labeled amino acid and heavy water.">
<meta name="twitter:image" content="https://github.com/ed-lau/jcast/images/riana_logo.png">
<meta name="twitter:creator" content="@LauLabColorado">
<meta name="twitter:image-height" content="1034">
<meta name="twitter:image-width" content="2550">
<meta name="twitter:image:alt" content="RIANA - Relative Isotope Abundance Analyzer">
<meta name="twitter:card" content="summary_large_image">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">RIANA v0.8.1</span>
  </a>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html">Home</a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./install.html">Quick Start</a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./about.html">Documentation</a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./change.html">Change Log</a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="./faqs.html" aria-current="page">FAQs</a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#faqs" id="toc-faqs" class="nav-link active" data-scroll-target="#faqs">FAQs</a>
  <ul class="collapse">
  <li><a href="#integrate" id="toc-integrate" class="nav-link" data-scroll-target="#integrate">Integrate</a></li>
  <li><a href="#fit" id="toc-fit" class="nav-link" data-scroll-target="#fit">Fit</a></li>
  <li><a href="#general" id="toc-general" class="nav-link" data-scroll-target="#general">General</a></li>
  </ul></li>
  </ul>
<div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/ed-lau/riana/issues/new" class="toc-action">Report an issue</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Frequently Asked Questions</h1>
</div>

<div>
  <div class="description">
    RIANA enables protein turnover data analysis from mass spectrometry-based proteomics experiments. Supports multiple labeling strategies, including labeled amino acid and heavy water.
  </div>
</div>


<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Edward Lau </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">May 1, 2025</p>
    </div>
  </div>
    
  </div>
  
<div>
  <div class="abstract">
    <div class="abstract-title">Abstract</div>
    This page provides the documentation for the RIANA package, which is a tool for the analysis of stable isotope labeling experiments for protein turnover measurements.
  </div>
</div>

</header>

<section id="faqs" class="level2">
<h2 class="anchored" data-anchor-id="faqs">FAQs</h2>
<section id="integrate" class="level3">
<h3 class="anchored" data-anchor-id="integrate">Integrate</h3>
<section id="q-what-does-the--d---mass_defect-argument-do-for-the-integrate-function" class="level4">
<h4 class="anchored" data-anchor-id="q-what-does-the--d---mass_defect-argument-do-for-the-integrate-function">Q: What does the <code>-D --mass_defect</code> argument do for the integrate function?</h4>
<p>A: The <code>-D</code> flag specifies the accurate mass difference between each unit isotopomer and can be set to <code>D</code>, <code>C13</code>, or <code>SILAC</code>, which will dictate the mass difference across each successive peak to be 1.0063 (mass difference of deuterium and protium), 1.003 (mass difference of <sup>13</sup>C and <sup>12</sup>C), and 1.001 (custom SILAC mass), respectively. This is set so that RIANA can find the correct accurate mass for integration when multiple atom centers are added. E.g., if deuterium labeled amino acids are used, e.g., D<sup>3</sup>-leucine, the argument should be set to <code>-D D</code> and the integrated isotopomer argument set to <code>-i 0, 3</code>. Alternatively if <sup>13</sup>C<sub>6</sub>-lysines, the arguments should be <code>-D C13</code> and <code>i 0,6</code>.</p>
</section>
</section>
<section id="fit" class="level3">
<h3 class="anchored" data-anchor-id="fit">Fit</h3>
<section id="q-what-are-the-kinetic-models-being-specified-by-the-different-fit-options" class="level4">
<h4 class="anchored" data-anchor-id="q-what-are-the-kinetic-models-being-specified-by-the-different-fit-options">Q: What are the kinetic models being specified by the different fit options?</h4>
<p>A: The <code>simple</code>, <code>guan</code>, and <code>fornasiero</code> options for the model argument refer to a simple exponential (one-pool) model, a two-compartment model, and a two-compartment model that includes label reutilization. See the <a href="./about.html">Documentation</a> page for details.</p>
</section>
<section id="q-what-does-the--l---label-argument-do" class="level4">
<h4 class="anchored" data-anchor-id="q-what-does-the--l---label-argument-do">Q: What does the <code>-l --label</code> argument do?</h4>
<p>A: The <code>--label</code> argument determines how the fraction of new proteins at a particular labeling time point is calculated from the relative abundance of isotopomers. The option <code>aa</code> assumes that the first peaks of light or heavy amino acids are used for calculations (e.g., m0, m6 for <sup>13</sup>C<sub>6</sub>-lysine) and hence the initial pre-label fraction of the light peak (m0) is assumed to be 1. The option <code>hw</code> assumes a heavy water type labeling experiment where the entire isotopomer cluster of the peptide is integrated (e.g., m0, m1, m2, m3, m4, m5), and hence the pre-labeling fraction of the m0 peak is calculated from peptide relative isotopomer abundance based on naturally occurring isotopes.</p>
<p><strong>To perform cell culture analysis:</strong> Use <code>-l hw_cell</code> to use the labeling sites calibrated for human AC6 cells (see Alamillo et al.&nbsp;bioRxiv 2025 for details). To alter the heavy water per-residue labeling sites for other cell types or organisms, modify the <code>params.py</code> file directly and run from the modified copy of RIANA.</p>
</section>
<section id="q-how-are-the-plotted-curves-separated-to-the-curves_fast_fit-curves_mid_fit-and-curve_slow_fit-directories" class="level4">
<h4 class="anchored" data-anchor-id="q-how-are-the-plotted-curves-separated-to-the-curves_fast_fit-curves_mid_fit-and-curve_slow_fit-directories">Q: How are the plotted curves separated to the curves_fast_fit, curves_mid_fit, and curve_slow_fit directories?</h4>
<p>A: The directories are intended to facilitate navigation only rather than a rigorous indication of data quality. Plots for fitted curves with k &lt; 0.01 are in the “slow” directory, 0.01 &lt;= k &lt;= 1 in the “mid” directory, and 1 &lt; k in the “fast” directory. Plots for fitted curves with R<sup>2</sup> &gt;= 0.5 are in the “fit” directory, and those with R<sup>2</sup> &lt; 0.5 are in the “poor” directory.</p>
</section>
<section id="q-does-riana-fit-perform-protein-level-curve-fitting" class="level4">
<h4 class="anchored" data-anchor-id="q-does-riana-fit-perform-protein-level-curve-fitting">Q: Does RIANA fit perform protein level curve fitting?</h4>
<p>A: RIANA currently performs curve fitting at the peptide level only. Users can perform protein-level curve fitting using the <code>{sample}_riana.txt</code> integration result files. Alternatively, protein-level rate constants can be derived by taking a median or harmonic mean of the peptide-level rate constants with similar results, as we showed in our recent <a href="https://doi.org/10.1016/j.mcpro.2022.100252">publication</a>.</p>
</section>
<section id="q-how-do-i-retrieve-half-life-information-on-the-peptide-and-protein" class="level4">
<h4 class="anchored" data-anchor-id="q-how-do-i-retrieve-half-life-information-on-the-peptide-and-protein">Q: How do I retrieve half-life information on the peptide and protein?</h4>
<p>A: The half-life of a peptide or protein can be calculated from the output <code>k</code> parameter using the equation:</p>
<p><span class="math display">\[ t_{1/2} = ln(2)/k \]</span></p>
</section>
<section id="q-how-do-i-use-limited-mass-isotopomers-or-rule-based-mass-isotopomer-selection-for-fractional-synthesis-rate-calculation" class="level4">
<h4 class="anchored" data-anchor-id="q-how-do-i-use-limited-mass-isotopomers-or-rule-based-mass-isotopomer-selection-for-fractional-synthesis-rate-calculation">Q: How do I use limited mass isotopomers or rule-based mass isotopomer selection for fractional synthesis rate calculation?</h4>
<p>A: This can be modified under <code>-f --fs</code>. The default option is m0/mA <code>m0_mA</code> and can be changed to <code>m0_m1</code> or <code>Auto</code> - please see Currie et al.&nbsp;J Proteome Res 2025 for details.</p>
</section>
</section>
<section id="general" class="level3">
<h3 class="anchored" data-anchor-id="general">General</h3>
<section id="q-how-can-i-contribute-to-riana" class="level4">
<h4 class="anchored" data-anchor-id="q-how-can-i-contribute-to-riana">Q: How can I contribute to RIANA?</h4>
<p>A: Contact us at [edward.lau@cuanschutz.edu]</p>


</section>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      let href = ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">© Copyright 2021–2025 <a href="https://laulab.net">Lau Lab Colorado</a></div>   
    <div class="nav-footer-right">This page is built with <a href="https://quarto.org/">Quarto</a>.</div>
  </div>
</footer>



</body></html>